### PR TITLE
Add people demo from #1388

### PR DESCRIPTION
**Pull request by andrewiggins**
Add the amazing people demo @phaux linked to in #1388 to our repo to help us catch errors and inefficiencies in children diffing sooner.

The original in written in TypeScript. I just copied it over as is and am using the new babel-typescript preset to remove the types.

I did convert the original css files to scss in order to scope the styles to only the people demo HTML and not the entire app.

### CODE REVIEW

File: demo/webpack.config.js [hunk 1]
```diff
@@ -35,12 +60,12 @@ module.exports = {
 							modules: false,
 							loose: true
 						}],
-						require.resolve('@babel/preset-react'),
+						[require.resolve('@babel/preset-react')],
```
Comment (andrewiggins) [inline demo/webpack.config.js]: I added these here just cuz I found it easier to read when al the plugins and presets lined up in a nice column. yay alignment!

File: demo/package.json [hunk 2]
```diff
@@ -24,6 +27,10 @@
   "dependencies": {
     "d3-scale": "^1.0.7",
     "d3-selection": "^1.2.0",
-    "preact-render-to-string": "^4.1.0"
+    "mobx": "^5.9.0",
+    "mobx-preact": "^3.0.0",
+    "mobx-state-tree": "^3.12.2",
+    "preact-render-to-string": "^4.1.0",
```
Comment (marvinhagemeister) [inline demo/package.json]: We have a `next` tag for `preact-render-to-string` which points to 5.x and supports ceviche :+1:

File: demo/index.js [hunk 3]
```diff
@@ -21,6 +22,9 @@ if (!isBenchmark) {
 	initDevTools();
 }
 
+// mobx-state-tree fix
+window.setImmediate = setTimeout;
```
Comment (marvinhagemeister) [inline demo/index.js]: @mweststrate Are there any particular reasons to use `setImmediate` over `setTimeout`?

Comment (developit) [reply demo/index.js]: Also interested to hear why this was used. Looks like it's here:
https://github.com/mobxjs/mobx-state-tree/blob/d3e24ab5f89e0e197e2b8594a6de774f3e739351/packages/mobx-state-tree/src/core/flow.ts#L117

**Comment by marvinhagemeister** [comment]
FYI: Had to upgrade `webpack` and remove the `babel-pgramatic-jsx` plugin to get it to work for me.

**Pull request merged**